src/map.h

```diff
@@ -114,35 +114,32 @@ struct Voxel {
 		this->instance_data = instance_data;
 	}
 
-	uint16 fence; ///< 4 bits used for each edge. EDGE_NE..EDGE_NW. Gives the fence type or FENCE_TYPE_INVALID if the voxel has no fence for this edge.
+	/**
+	 * Fences of the voxel. @see FenceType
+	 * - bit  0.. 3: Fence type of the NE edge.
+	 * - bit  4.. 7: Fence type of the SE edge.
+	 * - bit  8..11: Fence type of the SW edge.
+	 * - bit 12..15: Fence type of the NW edge.
+	 */
+	uint16 fences;
 
 	/**
-	 * Set the fence type of one of the four edges.
-	 * @param edge The edge to modify EDGE_NE..EDGE_NW
-	 * @return The fence type or FENCE_TYPE_INVALID if the edge has no fence
-	 * @note For steep slope all fences are stored in the top voxel. For non-steep
-	 * slope that have both corners raised, the fence type is stored in the voxel
-	 * above. This method will get what is stored in this voxel. You need to make
-	 * sure to call this method for the right voxel in the voxel stack.
+	 * Get the fences of the voxel. Use #GetFenceType and #SetFenceType for further manipulation
+	 * of the fence data.
+	 * @return Type of the fences at each edge.
 	 */
-	inline FenceType GetFenceType(TileEdge edge) const
+	inline uint16 GetFences() const
 	{
-		return (FenceType)GB(this->fence, (edge & 3) * 4, 4);
+		return this->fences;
 	}
 
 	/**
-	 * Set the fence type of one of the four edges.
-	 * @param edge The edge to modify EDGE_NE..EDGE_NW
-	 * @param fence_type The fence type to use or FENCE_TYPE_INVALID to disable fence of this edge
-	 * @note For steep slope all fences should be stored in the top voxel. For non-steep
-	 * slope that have both corners raised, the fence type should be stored in the voxel
-	 * above. This method will set the fence type of the given voxel. You need to make
-	 * sure to call this method for the right voxel in the voxel stack.
+	 * Set all the fences of the voxel.
+	 * @param fences Type of the fences at each edge.
 	 */
-	inline void SetFenceType(TileEdge edge, FenceType fence_type)
+	inline uint16 SetFences(uint16 fences)
 	{
-		assert(fence_type < FENCE_TYPE_COUNT || fence_type == FENCE_TYPE_INVALID);
-		SB(this->fence, (edge & 3) * 4, 4, fence_type & 0xF);
+		return this->fences = fences;
```

Comment:
Set* functions don't normally return a value, and at a first glance this looks like a broken == comparison